readthedocs/rtd_tests/tests/test_resolver.py

```diff
@@ -488,6 +488,19 @@ def test_resolver_public_domain_overrides(self):
             url = resolve(project=self.pip, private=False)
             self.assertEqual(url, 'http://docs.foobar.com/en/latest/')
 
+    @override_settings(
+        PRODUCTION_DOMAIN='readthedocs.org',
+        PUBLIC_DOMAIN='readthedocs.io',
+        PUBLIC_DOMAIN_USES_HTTPS=True,
```

Comment:
Instead of override this outside the method you may use `with` inside the method and test the same cases you are testing now, but using `PUBLIC_DOMAIN_USES_HTTPS=False` and test that in that case both are `http`

Reply:
That's a pretty good idea.